### PR TITLE
Split sanity workflow matrix into four workflows

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Split sanity workflow matrix into four workflows, tests are too unstable right now this will allow us to re-run them one at a time
| Type?             | improvement
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | ~
| How to test?      | CI tests green
| Possible impacts? | ~

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/23861)

### CODE REVIEW

**Review by PululuK — COMMENTED** [review]
@jolelievre  
it seems there is no way to factor or reuse workflows 😔
https://github.community/t/reusing-sharing-inheriting-steps-between-jobs-declarations/16851

**Comment by jolelievre** [comment]
> @jolelievre
> it seems there is no way to factor or reuse workflows 😔
> https://github.community/t/reusing-sharing-inheriting-steps-between-jobs-declarations/16851

Yep we didn't find a better solution for now It's not clean I agree but it should save us time, I hope!

**Review by PululuK — DISMISSED** [review]

**Review by boubkerbribri — APPROVED** [review]

File: .github/workflows/sanity-72.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
+name: UI tests - PHP 7.2
+
+on:
+  pull_request:
+    types: [opened,reopened,synchronize]
+
```
Comment (boubkerbribri) [inline .github/workflows/sanity-72.yml:6]:
```suggestion
on:
  pull_request:
    types: [opened,reopened,synchronize]
env: 
  php: '7.2'
  
```

File: .github/workflows/sanity-72.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    strategy:
+      matrix:
+        php: [ '7.2' ]
+      fail-fast: false
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup Apache
+        run: sudo bash .github/workflows/sanity/setup-apache.sh ${{ github.workspace }} ${{ matrix.php }}
```
Comment (boubkerbribri) [inline .github/workflows/sanity-72.yml:21]:
```suggestion
        run: sudo bash .github/workflows/sanity/setup-apache.sh ${{ github.workspace }} ${{ env.php }}
```

File: .github/workflows/sanity-72.yml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
```
Comment (boubkerbribri) [inline .github/workflows/sanity-72.yml:26]:
```suggestion
          php-version: ${{ env.php }}
```

File: .github/workflows/sanity-72.yml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+jobs:
+
+  sanity:
+    runs-on: ubuntu-18.04
+    name: Sanity - PHP 7.2
+    strategy:
+      matrix:
+        php: [ '7.2' ]
+      fail-fast: false
```
Comment (boubkerbribri) [inline .github/workflows/sanity-72.yml:15]:
```suggestion
```

**Review by boubkerbribri — COMMENTED** [review]
Little suggestion : Maybe we don't have to use Matrix anymore

**Review by PululuK — APPROVED** [review]
See @boubkerbribri  feedback's and it is ok for me 👍🏿

**Pull request merged**